### PR TITLE
GUI based Orders are almost done.

### PR DESCRIPTION
**Pull request by Vinyzu**
- Refractored some of the Code (Structure)
- Added Ordering Menu Selection Screen
- Added Order Check
- Implemented better Colors
- Added Quality of Life Buttons

### CODE REVIEW

**Comment by koviubi56** [comment]
Could you add `disallow_subclassing_any = false` to mypy's config (`pyproject.toml`) to make it happy?

File: src/sincere_singularities/modules/order.py [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+            colour=DISNAKE_COLORS.get(self.restaurant.icon, disnake.Color.random()),
+        )
+        # Adding an Empty Field for better formatting
+        embed.add_field(" ", " ")
+        # Adding Already Added Menu Items
+        for menu_name, menu_items in self.order.foods.items():
+            embed.add_field(
+                name=f"Added {menu_name} Items",
+                value="\n".join(
+                    [f"- `{item_name}`: {menu_items.count(item_name)}" for item_name in list(set(menu_items))]
```
Comment (koviubi56) [inline src/sincere_singularities/modules/order.py]:
```suggestion
                    f"- `{item_name}`: {menu_items.count(item_name)}" for item_name in set(menu_items)
```

File: src/sincere_singularities/modules/order.py [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+
+
+@dataclass(unsafe_hash=True)
+class CustomerInfo:
+    """The Dataclass Containing Information added in the Customer Information Section."""
+
+    name: str = ""
+    address: str = ""
+    delivery_time: str = ""
+    extra_information: str = ""
```
Comment (koviubi56) [inline src/sincere_singularities/modules/order.py]: I'm not sure why it has `unsafe_hash` and why it has defaults.

**Review by koviubi56 — CHANGES_REQUESTED** [review]
If you can fix these then it's ok

File: src/sincere_singularities/modules/order.py (hunk 2 above)
Comment (Vinyzu) [reply src/sincere_singularities/modules/order.py]: Switched to `frozen=True` and removed defaults

**Review by koviubi56 — APPROVED** [review]
LGTM (after you fix the formatting)

**Pull request merged**